### PR TITLE
SWORD2: Support language auto-detection

### PR DESCRIPTION
**Pull request by orgads**
Detect English, German and Hebrew (fan translation).

### CODE REVIEW

**Comment by orgads** [comment]
More complicated than I'd expect :)

Not sure if it's worth it.

CC @ShaharAriel

**Comment by bluegr** [comment]
@orgads please fix the conflicts, thanks!

**Comment by orgads** [comment]
I'm on vacation this week. Will fix it next week, sorry.

**Comment by orgads** [comment]
Done, and improved to detect also English.

File: engines/sword2/detection_internal.h [hunk 1]
```diff
@@ -76,25 +87,37 @@ static DetectedGames detectGamesImpl(const Common::FSList &fslist, bool recursio
...
-				// Match found, add to list of candidates, then abort inner loop.
-				DetectedGame game = DetectedGame("sword2", g->gameid, g->description,
-												 Common::UNK_LANG, g->platform);
-				game.setGUIOptions(GUIO2(GUIO_NOMIDI, GUIO_NOASPECT));
-
-				detectedGames.push_back(game);
-				break;
+				found = true;
 			}
+			else if (file->getName().equalsIgnoreCase("text.clu")) {
```
Comment (bluegr) [inline engines/sword2/detection_internal.h:105]:
```suggestion
			} else if (file->getName().equalsIgnoreCase("text.clu")) {
```

File: engines/sword2/detection_internal.h [hunk 2]
```diff
@@ -76,25 +87,37 @@ static DetectedGames detectGamesImpl(const Common::FSList &fslist, bool recursio
...
+				const int size = tmp.size();
+				for (const Sword2::GameLanguage &lang : Sword2::languages) {
+					if (size == lang.size) {
+						game.language = lang.language;
+						break;
+					}
+				}
+			}
+		}
+		if (found) {
```
Comment (bluegr) [inline engines/sword2/detection_internal.h:117]: Spacing before the if statement

File: engines/sword2/detection_internal.h (hunk 1 above)
Comment (orgads) [reply engines/sword2/detection_internal.h]: Done

File: engines/sword2/detection_internal.h (hunk 2 above)
Comment (orgads) [reply engines/sword2/detection_internal.h:117]: Done

**Comment by sev-** [comment]
This could be obsoleted by #4243

**Comment by bluegr** [comment]
Since this is superseded by #4243, there's no point in extending the existing detection code. Closing in favor of the other PR

**Pull request closed without merging**